projects/plugins/crm/src/entities/class-invoice.php

```diff
@@ -0,0 +1,363 @@
+	 */
+	public $currency = '';
+
+	/**
+	 * The pay via property of the object
+	 *
+	 * @var int
+	 */
+	public $pay_via = -1;
+
+	/**
+	 * The logo url of the object
+	 *
+	 * @var string
+	 */
+	public $logo_url = '';
+
+	/**
+	 * The address to objtype property of the object
+	 *
+	 * @var int
+	 */
+	public $address_to_objtype = -1;
+
+	/**
+	 * The addressed from property of the object
+	 *
+	 * @var string
+	 */
+	public $addressed_from = '';
+
+	/**
+	 * The addressed to property of the object
+	 *
+	 * @var string
+	 */
+	public $addressed_to = '';
+
+	/**
+	 * The allow partial property of the object
+	 *
+	 * @var bool
+	 */
+	public $allow_partial = false;
+
+	/**
+	 * The allow tip property of the object
+	 *
+	 * @var bool
+	 */
+	public $allow_tip = false;
+
+	/**
+	 * The send attachments property of the object
+	 *
+	 * @var bool
+	 */
+	public $send_attachments = false;
+
+	/**
+	 * The hours or quantity property of the object
+	 *
+	 * @var bool
+	 */
+	public $hours_or_quantity = true;
+
+	/**
+	 * The date property of the object
+	 *
+	 * @var int
+	 */
+	public $date = -1;
+
+	/**
+	 * The due date property of the object
+	 *
+	 * @var int
+	 */
+	public $due_date = -1;
+
+	/**
+	 * The paid date property of the object
+	 *
+	 * @var int
+	 */
+	public $paid_date = -1;
+
+	/**
+	 * The hash viewed property of the object
+	 *
+	 * @var int
+	 */
+	public $hash_viewed = -1;
+
+	/**
+	 * The hash viewed count property of the object
+	 *
+	 * @var int
+	 */
+	public $hash_viewed_count = 0;
+
+	/**
+	 * The portal viewed property of the object
+	 *
+	 * @var int
+	 */
+	public $portal_viewed = -1;
+
+	/**
+	 * The portal viewed count property of the object
+	 *
+	 * @var int
+	 */
+	public $portal_viewed_count = 0;
+
+	/**
+	 * The net property of the object
+	 *
+	 * @var float
+	 */
+	public $net = 0.0;
+
+	/**
+	 * The discount property of the object
+	 *
+	 * @var float
+	 */
+	public $discount = 0.0;
+
+	/**
+	 * The discount type property of the object
+	 *
+	 * @var string
+	 */
+	public $discount_type = '';
+
+	/**
+	 * The shipping property of the object
+	 *
+	 * @var float
+	 */
+	public $shipping = 0.0;
+
+	/**
+	 * The shipping taxes property of the object
+	 *
+	 * @var string
+	 */
+	public $shipping_taxes = '';
+
+	/**
+	 * The shipping tax property of the object
+	 *
+	 * @var float
+	 */
+	public $shipping_tax = 0.0;
+
+	/**
+	 * The taxes property of the object
+	 *
+	 * @var string
+	 */
+	public $taxes = '';
+
+	/**
+	 * The tax property of the object
+	 *
+	 * @var float
+	 */
+	public $tax = 0.0;
+
+	/**
+	 * The total property of the object
+	 *
+	 * @var float
+	 */
+	public $total = 0.0;
+
+	/**
+	 * The last updated property of the object
+	 *
+	 * @var int
+	 */
+	public $lastupdated = -1;
+
+	/**
+	 * The created property of the object
+	 *
+	 * @var int
+	 */
+	public $created = -1;
+
+	/**
+	 * DB field name mapping. db_field => model_field.
+	 *
+	 * @var array
+	 */
+	private $field_map = array(
```

Comment:
This will be extracted to a Factory (Invoice_Factory) or Data Mapper (Invoice_Data_Mapper) class.